### ISSUE TITLE
Function gets called twice

### ISSUE THREAD

**Issue by romuye123** [marcoschwartz/aREST#142]
rest.function("status",status);

**Comment by marcoschwartz** [comment]
Hello,

This might happen when using aREST in your browser, as browsers usually send several requests to the server. If you are using it from a web app however this won't happen. Is this an issue for you?

**Comment by romuye123** [comment]
Yes i am accessing the URL from web browser. Did not tried from app yet, will try and let you know the feedback.

**Comment by adadgio** [comment]
Is there any update on this issue ? I think it's still quite a problem for a Rest (;-)) like API not to be called properly from browser. Solution, skip `OPTIONS` request in the source code maybe ?

**Comment by marcoschwartz** [comment]
Still working on it - I found the source of the problem but it's hard to fix without breaking the compatibility with older boards like the Arduino Uno.

**Comment by pta-** [comment]
Found out that in chrome it come twice, but in edge only once. First time when I liked edge for a short time :)

**Comment by JLartigaud** [comment]
Hello @marcoschwartz, maybe you could create a branch for a working version with earlier versions ? Really impressed by the aRest library btw 👍

**Comment by pkerspe** [comment]
I could "solve" the issue of duplicate function calls in Chrome by registering a second function for requests to "favicon.ico":
`rest.function("favicon.ico", favIcon);`
Just create a handler function that returns just anything for this call.
Once I set this up I do not get duplicate calls anymore to my other rest endpoints.
Maybe this helps one or the other

**Comment by JLartigaud** [comment]
Haha, excellent! Didn't think about the favicon chrome call... Thank you for this really useful hint 👍

**Comment by pkerspe** [comment]
I created a fork that handles OPTIONS calls by directly sending a hopefully proper OPTIONS request.
Use at your own risk :-)
https://github.com/pkerspe/aREST

I changed the aRest.h file, basically added a function for the OPTIONS headers in line 309:

`void send_options_http_headers(){
	addToBuffer(F("HTTP/1.1 204 OK\r\nAccess-Control-Allow-Origin: *\r\nAllow: POST, GET, PUT\r\nAccess-Control-Allow-Methods: POST, GET, PUT\r\n\r\n"));
}`

and a check for OPTIONS request type in line 864:
`	if (answer.startsWith("OPTIONS")){
		Serial.println("Got OPTIONS call, will quit");
		state = 'x';
		command = 'o';
		return;
	}`

at at last a check for the new command type 'o' (which I added) in line 1096:
`  if (command == 'o'){
  	resetBuffer();
  	send_options_http_headers();
  	return true;
  }`

Thats basically, since I only use direct http rest api calls I did not test if it has any side effects on the rest to the module, unfortunately the inline doc is not very good, so I had to guess a bit if what I was doing was ok :-)

**Comment by marcoschwartz** [comment]
@pkerspe Thanks for your work! Indeed I know it's a nasty bug, basically I am mostly using it with direct REST calls from apps in Node.js so the favicon issue doesn't appear :)

Could you actually submit a pull request with your changes? That would be awesome!

**Comment by nenadvkc** [comment]
@marcoschwartz  The function still gets called twice when using aREST in the browser, is there a solution for this problem? is code 
 that @pkerspe made, merged?

**Comment by nenadvkc** [comment]
@marcoschwartz  Actually  rest.function("favicon.ico", favIcon); is working for me, sry :)
